### DISCUSSION TITLE
Is it secure to store encrypted keepas database on google drive?

### DISCUSSION THREAD

**Discussion by infinitieunique** [Kunzisoft/KeePassDX#1909] [Q&A] [answered]
Does the DB password ever get sent to the google servers if I use KeepasDX app, store the DB itself on google drive and decrypt it opening it as though it was a local storage (using android's built-in file picker and choosing /myGdrive/database.kdbx as directory)? Or is decryption done entirely on client side?

In case it only ever gets decrypted in RAM then I am good, but if the passphrase leaves the ram and interacts with the disk (in my case gdrive filesystem) then I am screwed.

Who knows, maybe it, for example, gets decrypted and then written down to the disc temporarily and then gets encrypted again when you save changes and lock it...

**Comment by J-Jamet** [comment, marked as answer]
What is sent by the file save stream is only encrypted. So for any file manager, there's no choice: it's only the encrypted data that's in transit. The only problem with these file managers is synchronization and storage conflicts, but your main credential is never sent to the storage access framework.
https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync#native-synchronization
